instrumentation/net/http/otelhttp/config.go

```diff
@@ -141,6 +143,14 @@ func WithSpanOptions(opts ...trace.SpanStartOption) Option {
 	})
 }
 
+// WithMetricAttributes adds a function that will be called to add attributes
+// to the request metrics.
+func WithMetricAttributes(metricAttributes func(operation string, r *http.Request, statusCode int) []attribute.KeyValue) Option {
```

Comment:
Should this define a structure to be passed to the function rather than a set of parameters?  This approach doesn't leave any room for providing additional information to the attribute functions later.